Add surface wind vector plots recipe and related function

plutowing opened the pull request:
<!-- Thanks for contributing! Please add a short description of your change, and link to an issue, e.g. "Fixes #123" -->
Fixes #1483
The PR creates surface wind vector plots using U10 and V10 components. This PR works and a figure is shown below for the sample plot.
### Contribution checklist

Aim to have all relevant checks ticked off before merging. See the [developer's guide](https://metoffice.github.io/CSET/contributing/) for more detail.

- [ ] Documentation has been updated to reflect change.
- [x] New code has tests, and affected old tests have been updated.
- [ ] All tests and CI checks pass.
- [x] Ensured the pull request title is descriptive.
- [ ] Conda lock files have been updated if dependencies have changed.
- [x] Attributed any Generative AI, such as GitHub Copilot, used in this PR.
- [x] Marked the PR as ready to review.
![pangu_surface_vector_plot_of_eastward_wind_and_northward_wind_6 0](https://github.com/user-attachments/assets/88c6b210-1075-4d3d-8a5d-64310ab0c89e)

jfrost-mo commented:
The other thing this pull request needs is a few tests. Have a look at the existing plotting tests for some inspiration. To run them locally make sure you have [setup the development tooling first](https://metoffice.github.io/CSET/contributing/getting-started.html#setting-up-tooling).

https://github.com/MetOffice/CSET/blob/c786342584618f08abbe4c2984e05dd44a56378c/tests/operators/test_plots.py#L245-L259

plutowing commented on src/CSET/operators/plot.py:
```diff
@@ -1655,6 +1781,61 @@ def scatter_plot(
+        cube_v.coord(sequence_coordinate)
+    except iris.exceptions.CoordinateNotFoundError as err:
+        raise ValueError(f"Cube(v) must have a {sequence_coordinate} coordinate.") from err
+
```
added sequence coord checking

plutowing commented:
changes of vector plot include:

1. fixed some docstrings, 
2. arrow density depends on cube dimensions
3. added a test and a test data for vector plot (already tested run locally)
<img width="788" height="544" alt="image" src="https://github.com/user-attachments/assets/33fd9bc9-c2fc-4e5b-858e-8dc9b5c06f39" />

jfrost-mo reviewed (commented):
Looking good. I've put a few minor comments, but I'll address them myself.

jfrost-mo commented on tests/conftest.py:
```diff
@@ -83,6 +83,36 @@ def vertical_profile_cube(vertical_profile_cube_readonly):
+    # filter the cubes using the constraints.
+    cube_u = filters.filter_cubes(in_cubes, combined_constraint_u)
+    cube_v = filters.filter_cubes(in_cubes, combined_constraint_v)
+    return [cube_u, cube_v]
```
Done in 2af8a9a9861bba918f6c7bea1b4f9286c1aeb509

jfrost-mo commented on src/CSET/operators/plot.py:
```diff
@@ -1809,6 +1806,10 @@ def vector_plot(
 
+    # check that the u and v cubes have the same sequence coordinate.
+    if cube_u.coord(sequence_coordinate) != cube_v.coord(sequence_coordinate):
+        raise ValueError(f"Cube(u) and Cube (v) must have the same {sequence_coordinate} coordinate.")
```
Done in b7c018991c3f46ae63be73b17a338827966121dd

jfrost-mo commented on src/CSET/operators/plot.py:
```diff
@@ -1809,6 +1806,10 @@ def vector_plot(
     except iris.exceptions.CoordinateNotFoundError as err:
         raise ValueError(f"Cube(v) must have a {sequence_coordinate} coordinate.") from err
 
+    # check that the u and v cubes have the same sequence coordinate.
```
Done in b7c018991c3f46ae63be73b17a338827966121dd

jfrost-mo commented:
Test coverage measurement: https://tmpweb.net/NFJuMGM48tNz/
